src/fixit/lsp.py

```diff
@@ -0,0 +1,162 @@
+# Copyright (c) Meta Platforms, Inc. and affiliates.
+#
+# This source code is licensed under the MIT license found in the
+# LICENSE file in the root directory of this source tree.
+
+from pathlib import Path
+from typing import Callable, Dict
+
+import pygls.uris as Uri
+
+from lsprotocol.types import (
+    Diagnostic,
+    DiagnosticSeverity,
+    DidChangeTextDocumentParams,
+    DidOpenTextDocumentParams,
+    DocumentFormattingParams,
+    Position,
+    Range,
+    TEXT_DOCUMENT_DID_CHANGE,
+    TEXT_DOCUMENT_DID_OPEN,
+    TEXT_DOCUMENT_FORMATTING,
+    TextEdit,
+)
+from pygls.server import LanguageServer
+
+from fixit import __version__
+from fixit.util import capture, debounce
+
+from .api import fixit_bytes
+from .config import generate_config
+from .ftypes import Config, LspOptions, Options
+
+
+class LSP:
+    """
+    Server for the Language Server Protocol.
+    Provides diagnostics as you type, and exposes a formatter.
+    https://microsoft.github.io/language-server-protocol/
+    """
+
+    def __init__(self, fixit_options: Options, lsp_options: LspOptions):
```

Comment:
A number of methods here need type annotations.

Reply:
Added return types